__tests__/functional/client-configuration.test.ts

```diff
@@ -64,6 +63,27 @@ an environmental variable named FAUNA_SECRET or pass it to the Client constructo
     expect(result.txn_time).toBeDefined();
   });
 
+  it("client allows txn time to be set", async () => {
+    const client = new Client({
+      endpoint: endpoints.local,
+      secret: "secret",
+    });
+    expect(client.last_txn_time).toBeUndefined();
+    const expectedTxnTime = new Date(Date.now());
+    client.last_txn_time = expectedTxnTime;
+    expect(client.last_txn_time).toBe(expectedTxnTime);
+    const addFiveMinutes = new Date(expectedTxnTime.getTime() + 5 * 60000);
+    client.last_txn_time = addFiveMinutes;
+    expect(client.last_txn_time).toBe(addFiveMinutes);
+    // setting txn time back in history should fail
+    expect(() => {
+      client.last_txn_time = expectedTxnTime;
+    }).toThrow();
+    expect(() => {
+      client.last_txn_time = undefined;
+    }).toThrow();
```

Comment:
I'm unsure about this bit, kinda feel like we should let customers "reset" it, but not sure if there's a reasonable use case for this or if we'd just expect them to init a new client. Question has been raised in Slack for broader input.

Reply:
> in general, no. I would say it’s exceptional enough that they could in that case create a new client like you say